mix.exs

```diff
@@ -42,7 +42,7 @@ defmodule Guardian.Mixfile do
 
   defp deps do
     [{:jose, "~> 1.6"},
-     {:poison, "~> 2.1"},
+     {:poison, ">= 1.3.0 and < 3.0.0"},
```

Comment:
So long as it's above 2 I'm pretty sure we're good. So long as we can smash JSON we're good. I don't think there is even a real requirement from Guardians side other than it be there since we're just doing string keys. I'd be happy to merge this but maybe we can just say >= 1.3.0?